flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/overlays/HadoopUserOverlay.java

```diff
@@ -57,24 +57,58 @@ public void configure(ContainerSpecification container) throws IOException {
 	}
 
 	public static Builder newBuilder() {
-		return new Builder();
+		// First check if we have Hadoop in the ClassPath. If not, we simply don't do anything.
```

Comment:
`First check if we have Hadoop in the ClassPath.` -> `First check if we have Hadoop specific dependencies in the ClassPath.` looks better to me